aggregation/src/lib.rs

```diff
@@ -0,0 +1,495 @@
+// Copyright (c) 2024 RISC Zero, Inc.
+//
+// All rights reserved.
+
+extern crate alloc;
+
+use alloc::vec::Vec;
+use core::{borrow::Borrow, fmt::Debug};
+
+use alloy_primitives::{Keccak256, B256};
+use alloy_sol_types::SolValue;
+use anyhow::{anyhow, bail, Error, Result};
+use risc0_binfmt::{tagged_struct, Digestible};
+use risc0_zkvm::{
+    sha,
+    sha::{Digest, Sha256, DIGEST_BYTES},
+    InnerReceipt, MaybePruned, Receipt, ReceiptClaim, VerifierContext,
+};
+use serde::{Deserialize, Serialize};
+
+alloy_sol_types::sol! {
+    /// Seal of the SetInclusionReceipt.
+    #[sol(all_derives)]
+    struct Seal {
+        /// Merkle path to the leaf.
+        bytes32[] path;
+        /// Root seal.
+        bytes root_seal;
+    }
+}
+
+/// A receipt for a claim that is part of a set of verified claims (i.e. an aggregation).
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[non_exhaustive]
+pub struct SetInclusionReceipt<Claim>
```

Comment:
This struct is never used by the guest, so I suppose it is used outside?

In that case, shouldn't the entire SetInclusionReceipt be behind the feature flag?
I guess it could make sense to separate it from the GuestInput/-Output:
- Maybe rename the input to SetBuilderInput or something similar.
-Put it into different files.

Reply:
Ya, let's go ahead and more this to a new file 👍 
3ae5e230be1a38d4ad1583ff707924ed5764086a